confgenerator/confgenerator.go

```diff
@@ -118,6 +162,88 @@ func unifiedConfigReader(input []byte) (unifiedConfig, error) {
 	return config, nil
 }
 
+func unifiedConfigWindowsReader(input []byte) (unifiedConfigWindows, error) {
+	config := unifiedConfigWindows{}
+	err := yaml.UnmarshalStrict(input, &config)
+	if err != nil {
+		return unifiedConfigWindows{}, err
+	}
+	return config, nil
+}
+
+func generateOtelConfig(metrics *otelMetrics, logsDir string) (string, error) {
+	hostMetricsList := []*otel.HostMetrics{}
+	stackdriverList := []*otel.Stackdriver{}
+	serviceList := []*otel.Service{}
+	receiverNameMap := make(map[string]string)
+	exporterNameMap := make(map[string]string)
+	if metrics.Service != nil {
+		hostmetricsReceiverFactories, err := extractOtelReceiverFactories(metrics.Receivers)
+		if err != nil {
+			return "", err
+		}
+		hostMetricsList, receiverNameMap, err = generateOtelReceivers(hostmetricsReceiverFactories, metrics.Service.Pipelines)
+		if err != nil {
+			return "", err
+		}
+		stackdriverList, exporterNameMap, err = generateOtelExporters(metrics.Exporters, metrics.Service.Pipelines)
+		if err != nil {
+			return "", err
+		}
+		serviceList, err = generateOtelServices(receiverNameMap, exporterNameMap, metrics.Service.Pipelines)
+	}
+	otelConfig, err := otel.GenerateOtelConfig(hostMetricsList, stackdriverList, serviceList)
+	if err != nil {
+		return "", err
+	}
+	fmt.Print(otelConfig)
+	return otelConfig, nil
+}
+
+func generateOtelServices(receiverNameMap map[string]string, exporterNameMap map[string]string, pipelines map[string]*otelPipeline) ([]*otel.Service, error) {
+	serviceList := []*otel.Service{}
+	var pipelineIDs []string
+	for p := range pipelines {
+		pipelineIDs = append(pipelineIDs, p)
+	}
+	sort.Strings(pipelineIDs)
+	for _, pID := range pipelineIDs {
+		p := pipelines[pID]
+		var pRecevieIDs []string
```

Comment:
i know i have typo somewhere. it showed on intellij, then it slipped away. i globally seached for "reci" and "Reci". Didn' expect it could be more wrong than that..